### DISCUSSION TITLE
Trying to understand what the approriate unit is for a rate metric, like bytes/s is

### DISCUSSION THREAD

**Discussion by leidegre** [microsoft/aspire#5495] [Q&A] [answered]
So, I have a gauge that reports bytes per second. I've set this up like so and I'm using OTLP.

```cs
s_meter.CreateObservableGauge<double>("msab.dems.storage.client.uploaded_speed", () =>
{
    double sum = 0;
    for (int i = 0; i < uploadSpeed.Length; i++)
    {
        sum += uploadSpeed[i];
    }
    return new Measurement<double>(sum / uploadSpeed.Length);
}, unit: "By/s"); // Aspire will convert OTLP units to "bytes_per_second"?
```

This ends up in Aspire and is presented like this, which is sort of fine but seems wrong given that it ran "per" and "seconds" together to form "Perseconds". I'm trying to understand what makes it do that.

![image](https://github.com/user-attachments/assets/176aec76-86a6-45bc-ad4e-1102cb958c72)

I've poked around in the code trying to figure out how this is supposed to work but any input on this would be much apricated. I just want the unit to be presented properly.

**Comment by davidfowl** [comment]
cc @samsp-msft @noahfalk

**Comment by noahfalk** [comment, marked as answer]
@samsp-msft - This looks like it may be a client side UI issue in the Aspire dashboard. The unit looks correct to me as defined in code and both dotnet-counters and OTel console exporter display it correctly:

dotnet-counters
```
Name                                         Current Value
[MyMeter]
    MyGauge (By/s)                                   1    
```

OTel console exporter
```
Resource associated with Metric:
    telemetry.sdk.name: opentelemetry
    telemetry.sdk.language: dotnet
    telemetry.sdk.version: 1.9.0
    service.name: unknown_service:Launcher

Metric Name: MyGauge, Unit: By/s, Meter: MyMeter
(2024-08-30T22:18:44.7770892Z, 2024-08-30T22:18:51.2528531Z] LongGauge
Value: 1
```

**Reply by davidfowl** [reply]
cc @JamesNK

**Comment by JamesNK** [comment]
Will be fixed with https://github.com/dotnet/aspire/pull/5506

You'll have to put up with badly formatted units until the next Aspire release.

**Reply by leidegre** [reply]
np, as long as gets better over time a will cope, thanks for taking the time but just to set expectations, when's the next release? what are we talking about, a couple of months?

**Reply by JamesNK** [reply]
Couple of months. 8.2 just shipped so it will be a little while until the next release
